packages/mobx-state-tree/__tests__/core/model.test.ts

```diff
@@ -17,152 +18,405 @@ test("it should call preProcessSnapshot with the correct argument", () => {
+                const date = new Date("2023-07-24T04:26:04.701Z")
+                const Model = types.model({
+                    foo: date
+                })
 
-            const modelSnapshot = getSnapshot(Model.create())
-            expect(modelSnapshot).toEqual({})
+                const modelSnapshot = getSnapshot(Model.create())
+                expect(modelSnapshot).toEqual({
+                    foo: 1690172764701
+                })
+            })
         })
     })
-    describe("when we provide three arguments to the function", () => {
-        test("the model gets the correct name", () => {
-            // @ts-ignore
-            const Model = types.model("name", {}, {})
+    describe("when a user defines a property using a complex type", () => {
+        describe('and that type is "types.map"', () => {
+            test("it sets the default value to an empty map", () => {
+                const Model = types.model({
+                    foo: types.map(types.string)
+                })
 
-            expect(Model.name).toBe("name")
+                const modelSnapshot = getSnapshot(Model.create())
+                expect(modelSnapshot).toEqual({
+                    foo: {}
+                })
+            })
         })
-        test("the model gets the correct properties", () => {
-            const Model = types.model(
-                "name",
-                {
-                    prop1: "prop1",
-                    prop2: 2
-                },
-                // @ts-ignore
-                {}
-            )
+        describe('and that type is "types.array"', () => {
+            test("it sets the default value to an empty array", () => {
+                const Model = types.model({
+                    foo: types.array(types.string)
+                })
 
-            const modelSnapshot = getSnapshot(Model.create())
-            expect(modelSnapshot).toEqual({
-                prop1: "prop1",
-                prop2: 2
+                const modelSnapshot = getSnapshot(Model.create())
+                expect(modelSnapshot).toEqual({
+                    foo: []
+                })
+            })
+        })
+        describe("and that type is another model", () => {
+            test("it sets the default value to the default of that model", () => {
+                const Todo = types.model({
+                    task: types.optional(types.string, "test")
+                })
+
+                const TodoStore = types.model("TodoStore", {
+                    todo1: types.optional(Todo, () => Todo.create())
+                })
+
+                const modelSnapshot = getSnapshot(TodoStore.create())
+                expect(modelSnapshot).toEqual({
+                    todo1: {
+                        task: "test"
+                    }
+                })
             })
         })
     })
+    describe("when a user defines a property using a function", () => {
+        // if (process.env.NODE_ENV === "production") {
```

Comment:
These tests fail, but I don't think they are intended to. My reading of [`toPropertiesObject`](https://github.com/mobxjs/mobx-state-tree/blob/b36e07a86a3b791b115d83983f6ee93143ac0dc5/packages/mobx-state-tree/src/types/complex-types/model.ts#L258) is that we only want to error out if `devMode()` comes back as truthy.

These test cases end up hitting the final `else` clause, which has a comment that reads `// WTF did you pass in mate?` - but really it's just that MST doesn't have a case to handle when a user tries to use a function as a property in production.

I don't think this is really a bug, because we expect developers to hit the failure in development before shipping to prod, but it's not technically accounted for.

@jamonholmgren - do you think we should:

1. Write a case in `toPropertiesObject` to handle functions/objects as props in production where MST doesn't fail, but sets the value to some kind of `undefined` or other falsy value?
2. Remove `devMode()` checks in the function and object cases, so that we can provide more useful errors in production environments
3. Something else?

Reply:
In the interest of getting this merged to add more test coverage [I removed the failing specs](https://github.com/mobxjs/mobx-state-tree/pull/2051/commits/9dff9372b4aa208734ffee1c9f623c596350d753). I don't think this is a major pitfall, just a nuance I was interested in. Perhaps we'll revisit!